analyzers/src/SonarAnalyzer.CSharp/Rules/LossOfFractionInDivision.cs

```diff
@@ -115,5 +143,38 @@ private static bool TryGetTypeFromAssignmentToFloatType(SyntaxNode division, Sem
             type = null;
             return false;
         }
+
+        private static SyntaxNode GetMostOuterTuple(SyntaxNode node) =>
+            node.Ancestors()
+                .TakeWhile(x => TupleExpressionSyntaxWrapper.IsInstance(x) || x.IsKind(SyntaxKind.Argument))
+                .LastOrDefault(x => TupleExpressionSyntaxWrapper.IsInstance(x));
+
+        private static int DivisionArgumentIndex(ImmutableArray<ArgumentSyntax> arguments, SyntaxNode division)
+        {
+            var divisionArgument = arguments.FirstOrDefault(x => x.Expression.Equals(division));
+            return arguments.IndexOf(divisionArgument);
```

Comment:
I think there is an overload of IndexOf which takes a predicate:
```suggestion
            return arguments.IndexOf(x => x.Expression.Equals(division));
```